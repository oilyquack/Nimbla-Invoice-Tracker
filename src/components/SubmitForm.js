import React from "react";

class SubmitForm extends React.Component {
  constructor() {
    super();

    this.state = {
      amount: "",
      payBy: "",
      customer: "",
      services: ""
    };

    this.submitHandler = this.submitHandler.bind(this);
    this.amountChange = this.amountChange.bind(this);
    this.payByChange = this.payByChange.bind(this);
    this.customerChange = this.customerChange.bind(this);
    this.servicesChange = this.servicesChange.bind(this);
  }

  amountChange(event) {
    this.setState({
      amount: event.target.value
    });
  }

  payByChange(event) {
    this.setState({
      payBy: event.target.value
    });
  }

  customerChange(event) {
    this.setState({
      customer: event.target.value
    });
  }

  servicesChange(event) {
    this.setState({
      services: event.target.value
    });
  }

  submitHandler(event) {
    event.preventDefault();

    const invoice = {
      id: this.props.invoiceID,
      amount: this.state.amount,
      payBy: this.state.payBy,
      customer: this.state.customer,
      services: this.state.services,
      paid: false
    };

    this.props.invoiceReceiver(invoice);

    this.setState({
      amount: "",
      payBy: "",
      customer: "",
      services: ""
    });
  }

  render() {
    return (
      <form className="app__form" onSubmit={this.submitHandler}>
        <label>Amount:</label>
        <input
          id="amount-input"
          placeholder="Â£"
          type="number"
          onChange={this.amountChange}
          value={this.state.amount}
        />

        <label>Pay By:</label>
        <input
          id="date-input"
          placeholder="Date"
          type="date"
          onChange={this.payByChange}
          value={this.state.payBy}
        />

        <label>Customer:</label>
        <input
          id="customer-input"
          placeholder="Customer"
          type="text"
          onChange={this.customerChange}
          value={this.state.customer}
        />

        <label>Services:</label>
        <textarea
          id="services-input"
          placeholder="Services"
          type="text"
          onChange={this.servicesChange}
          value={this.state.services}
        />

        <button type="submit">Submit</button>
      </form>
    );
  }
}

export default SubmitForm;
