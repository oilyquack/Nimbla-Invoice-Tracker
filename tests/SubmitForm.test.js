import React from "react";
import renderer from "react-test-renderer";
import SubmitForm from "../src/components/SubmitForm";

it("matches the snapshot", () => {
  const tree = renderer.create(<SubmitForm />);
  expect(tree).toMatchSnapshot();
});
