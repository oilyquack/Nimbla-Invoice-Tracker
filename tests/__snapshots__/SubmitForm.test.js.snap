// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<form
  className="app__form"
  onSubmit={[Function]}
>
  <label>
    Amount:
  </label>
  <input
    id="amount-input"
    onChange={[Function]}
    placeholder="Â£"
    type="number"
    value=""
  />
  <label>
    Pay By:
  </label>
  <input
    id="date-input"
    onChange={[Function]}
    placeholder="Date"
    type="date"
    value=""
  />
  <label>
    Customer:
  </label>
  <input
    id="customer-input"
    onChange={[Function]}
    placeholder="Customer"
    type="text"
    value=""
  />
  <label>
    Services:
  </label>
  <textarea
    id="services-input"
    onChange={[Function]}
    placeholder="Services"
    rows="5"
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
